### DISCUSSION TITLE
What's the right way to return a complex response?

### DISCUSSION THREAD

**Discussion by shi-yan** [rwf2/Rocket#2172] [Questions]
the document only shows simple scenarios IMO https://rocket.rs/v0.3/guide/responses/

Specifically, I don't know how error and a different content type can be returned.

for example

```rust
use rocket_contrib::Json;

#[derive(Serialize)]
struct Task { ... }

#[get("/todo")]
fn todo() -> Json<Task> { ... }
```

should todo() always succeed? when it fails, how can I return an error message? 

I'm currently doing this
```rust
#[post("/api/register")]
fn register() -> (Status, (ContentType, Vec<u8>)) 
{
    (
        Status::NotFound,
        (ContentType::HTML, "<p>Not Found!</p>".as_bytes().to_vec()),
    )
}

```

in order to assemble my own response. But each time, I need to provide status, and a content type as well as serialize content myself.

I feel this is very verbose.

**Comment by SergioBenitez** [comment]
That is very outdated documentation you're reading and completely not applicable to today's Rocket. The correct docs are at:

https://rocket.rs/v0.5-rc/guide/responses/

Also see the FAQ:

https://rocket.rs/v0.5-rc/guide/faq/#how-to
